kuma/javascript/src/active-banner.jsx

```diff
@@ -182,86 +179,80 @@ function Banner(props: BannerProps) {
-                // and the current user is not authenticated, return `null`
-                if (banner.authenticated && !userData.isAuthenticated) {
-                    return null;
-                }
+    const bannerId = userData.waffle.flags
+        ? BANNER_IDS.find(id => userData.waffle.flags[id] && !isEmbargoed(id))
+        : null;
 
-                // special case for the subscriptions banner as we need access
-                // to the username as it forms part of the `url`
-                if (banner.id === 'subscription_banner' && userData.username) {
-                    banner.url = `${window.mdn.wikiSiteUrl}/${locale}/profiles/${userData.username}/edit#subscription`;
-                }
+    if (!bannerId) {
+        return null;
+    }
+
+    switch (bannerId) {
+        case DEVELOPER_NEEDS_ID:
+            return (
+                <Banner
+                    id={bannerId}
+                    classname="developer-needs"
+                    title={gettext('MDN Survey')}
+                    copy={gettext(
+                        'Help us understand the top 10 needs of Web developers and designers.'
+                    )}
+                    cta={gettext('Take the survey')}
+                    url={
+                        'https://qsurvey.mozilla.com/s3/Developer-Needs-Assessment-2019'
+                    }
+                    newWindow
+                />
+            );
 
-                if (!isEmbargoed(banner.id)) {
-                    return <Banner {...banner} />;
-                }
+        case SUBSCRIPTION_ID:
+            if (!userData.isAuthenticated) {
+                return null;
             }
-        }
+
+            return (
+                <Banner
+                    id={bannerId}
+                    classname="mdn-subscriptions"
+                    title={gettext('Become a monthly supporter')}
+                    copy={
+                        <Interpolated
+                            id={gettext(
+                                'Support MDN with a $5 monthly subscription <learnMore/>.'
+                            )}
+                            learnMore={
+                                <a href={`/${locale}/payments/`}>
+                                    {gettext('Learn more')}
+                                </a>
+                            }
+                        />
+                    }
+                    cta={gettext('Subscribe')}
+                    url={`${
+                        window.mdn ? window.mdn.wikiSiteUrl : ''
+                    }/${locale}/profiles/${userData.username ||
+                        ''}/edit#subscription`}
+                    embargoDays={7}
+                />
+            );
+
+        default:
+            // TODO: throw it into Sentry once that's set-up for the frontend
+            // eslint-disable-next-line no-console
+            console.warn('Unknown banner for id', bannerId);
+            return null;
```

Comment:
I'd much rather this was `throw new Error(`Unknown banner '${bannerId}'`)` but I could go either way.